src/parsing.js

```diff
@@ -102,10 +102,83 @@
     return matchedUnits;
   }
 
+  function ignore(value) {
+    return function(input) {
+      var result = value(input);
+      if (result)
+        result[0] = undefined;
+      return result;
+    }
+  }
+
+  // Regex should be anchored with /^
+  function consumeToken(regex, string) {
+    var result = regex.exec(string);
+    if (result) {
+      result = regex.ignoreCase ? result[0].toLowerCase() : result[0];
+      return [result, string.substr(result.length)];
```

Comment:
Yeah my idea was either we change this to an object, or add a comment. I always lean toward making the code more expressive first, but if you would prefer a comment that's fine.